base/task.jl

```diff
@@ -102,12 +147,16 @@ function task_done_hook(t::Task)
                 active_repl_backend.in_eval
                 throwto(active_repl_backend.backend_task, result)
             end
-            let bt = catch_backtrace()
-                # run a new task to print the error for us
-                @schedule with_output_color(:red, STDERR) do io
-                    print(io, "ERROR (unhandled task failure): ")
-                    showerror(io, result, bt)
-                    println(io)
+            if collect_ex(t)
+                t.result=CollectedException(result, catch_backtrace())
```

Comment:
(in a separate PR) i think we could fix by adding a `finalizer` on every `Task`. in pseudo-code:
```julia
function task_finalizer(dead_task::Task)
  if !dead_task.handled && dead_task.state == :failed
    let result = dead_task.result, bt = dead_task.result_backtrace
        # run a new task to print the error for us
        @schedule with_output_color(:red, STDERR) do io
            print(io, "ERROR (unhandled task failure): ")
            showerror(io, result, bt)
            println(io)
        end
    end
  end
  nothing
end
tfin = cfunction(task_finalizer, Void, (Ref{Task},))
finalizer(newtask, tfin)
```